### ISSUE TITLE
Shapes should be replaced with more efficient versions used in carpet /draw

### ISSUE THREAD

**Issue by Firigion** [Ghoulboy78/World-edit-scarpet#40]
That's about it.

**Comment by Ghoulboy78** [comment]
k thx. ngl tho, it's not like the current carpet /draw commands are record-breakers for speed...

**Comment by Ghoulboy78** [comment]
but they're significantly better than the brush stuff.

**Comment by Firigion** [comment]
Yeah, the brush stuff is literally "check every block if it's inside, set it". And the check to see if it's inside is not even that good.

**Comment by altrisi** [comment]
Relates to gnembon/fabric-carpet#637.

Although that one is blocked by gnembon/fabric-carpet#618.

**Comment by Ghoulboy78** [comment]
This is a work in progress

**Comment by Ghoulboy78** [comment]
I'll add to #49 once gnembon/fabric-carpet#618 is accepted.

**Comment by Ghoulboy78** [comment]
ok now this depends on gnembon/fabric-carpet#742

**Comment by Ghoulboy78** [comment]
ok now this depends on gnembon/fabric-carpet#1154
